### DISCUSSION TITLE
FragmentType returns never

### DISCUSSION THREAD

**Discussion by matheusrocha89** [dotansimha/graphql-code-generator#8581] [Q&A] [answered]
Hi,
I am using `graphql-codegen` with the `client-preset` and `near-operation-file-preset`. All my fragments and queries are on files with extensions `.gql`. 

After configuring the presets I can generate the files with their types and so on. My problem is:
Following the documentation and examples, I saw that the property of the components should have a type declaration of: `FragmentType<typeof GraphqlFragmentDoc>` for example, from the generated file. But this returns a type never. So when I pass the object to this prop of my component the typescript complains about incompatible types.

<img width="793" alt="image" src="https://user-images.githubusercontent.com/3718366/200359187-6bace718-8cd4-4380-b80f-1ee0b5ddccc2.png">
<img width="652" alt="image" src="https://user-images.githubusercontent.com/3718366/200359285-a190062e-b58a-47df-a451-98f378fc8b2d.png">

Did somebody go through the same problem?

**Comment by charlypoly** [comment, marked as answer]
Hi @matheusrocha89,

Could you provide a simplified live example on GitHub or CodeSandbox?
Please know that the `client-preset` is not made to be used with `near-operation-file-preset`, this might be your issue here.

**Reply by matheusrocha89** [reply]
So I should use just one or the other? Can't use both together to have colocation files and Data Masking?

**Reply by matheusrocha89** [reply]
@charlypoly I created a repository with a small example using `create-react-app`. Here I am not using the `near-operation-file-preset`, just the `client` preset.

https://github.com/matheusrocha89/codege-fragment-example

**Reply by charlypoly** [reply]
> So I should use just one or the other? Can't use both together to have colocation files and Data Masking?

Yes, you can just use the `client-preset` and benefit from colocated GraphQL operation and Fragments definitions.

**Reply by charlypoly** [reply]
@matheusrocha89, if you are using the `client-preset`, you don't need to use the `typescript-*` packages (cf here: https://github.com/matheusrocha89/codege-fragment-example/blob/8c2c10dd881faf24fa66370203d9fb2b25a5a87e/codegen.ts#L10-L13)

I would encourage you to read our guide to get a better start with the `client-preset`: https://www.the-guild.dev/graphql/codegen/docs/guides/react-vue

**Reply by matheusrocha89** [reply]
Not even the apollo one to generate the hooks that I can use to remove the `useQuery` from apollo?
